### PR TITLE
Url capability

### LINKED ISSUE #344: :url capability
I would like to implement this, or connect with anyone who is working on it.

### PR DESCRIPTION
**Pull request by kpbarrett**
Implements :url capability ([section 8.8](https://tools.ietf.org/html/rfc6241#section-8.8)). Fixes #344. Also adds unit tests for \<delete-config> and \<validate>.

Notes:
- Can be disabled at config time with `-DENABLE_URL=OFF`.
- The \<copy-config> implementation allows the yang-library module to be copied out to URL. As a result, a copy out followed directly by a copy in on the same URL will fail. I think this is okay but wanted to point it out.
- Reviewers please double check thread safety, my test environment is minimal so I was not able to test it myself.

### CODE REVIEW

File: server/main.c [hunk 1]
```diff
@@ -1170,6 +1175,37 @@ np2srv_init_schemas(void)
...
+    /*
+     * Create a context for working with config XML in a standalone file.
+     *
+     * In standalone config files, the <config> node with namespace
+     * "urn:ietf:params:xml:ns:netconf:base:1.0" wraps xml data. We create a
+     * special schema module to define this to libyang so we can
+     * import/export these files. The overloading of the namespace prevents
+     * us from storing this module in our server context, so it must be kept
+     * in a new context.
+     */
```
Comment (michalvasko) [inline server/main.c]: This was the correct solution currently. However, I have added this `config` virtual node into `ietf-netconf` so you should have no problems using the existing context. Please, simplify your code this way.

File: server/op_copyconfig.c [hunk 2]
```diff
@@ -80,89 +111,115 @@ op_copyconfig(struct lyd_node *rpc, struct nc_session *ncs)
...
     }
-    /* TODO URL capability */
+
     ly_set_free(nodeset);
 
-    /* perform operation */
-    if (config) {
-        /* remove all the data from the models mentioned in the <config> ... */
+    if (source_is_ds && target_is_ds)
+    {
```
Comment (michalvasko) [inline server/op_copyconfig.c]: Please, always put the starting bracket on the same line as `if`, thanks.

**Comment by michalvasko** [comment]
Hi Kevin,
thanks a lot for your contribution, it is appreciated! Just address my comments and we can merge it.

Regards,
Michal

**Comment by kpbarrett** [comment]
Changes made, please review. Thanks!

**Comment by michalvasko** [comment]
Alright, thank you!

Regards,
Michal

**Pull request merged**